### ISSUE TITLE
(🐞) model config merging doesn't respect mro

### ISSUE THREAD

**Issue by KotlinIsland** [pydantic/pydantic#9992]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

in the example below, the output should be `True` in my opinion because `A` is first in the mro.

### Example Code

```Python
from pydantic import BaseModel, ConfigDict


class A(BaseModel):
    model_config = ConfigDict(strict=True)


class B(BaseModel):
    model_config = ConfigDict(strict=False)


class C(A, B):
    pass


print(C.model_config["strict"])  # False
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.8.2
        pydantic-core version: 2.20.1
          pydantic-core build: profile=release pgo=true
                 install path: C:\Users\AMONGUS\projects\test\.venv\Lib\site-packages\pydantic
               python version: 3.12.2 (tags/v3.12.2:6abddd9, Feb  6 2024, 21:26:36) [MSC v.1937 64 bit (AMD64)]
                     platform: Windows-10-10.0.19045-SP0
             related packages: typing_extensions-4.12.2
                       commit: unknown
```

**Comment by sydney-runkle** [comment]
Hmm, I'm not convinced this should change - this would be a breaking change. When we merge fields, we respect the settings on the last field with the highest priority...

**Comment by sydney-runkle** [comment]
Ah yes, we've established this pattern in our [docs](https://docs.pydantic.dev/latest/migration/#changes-to-config):

> When subclassing a model, the model_config attribute is inherited. This is helpful in the case where you'd like to use a base class with a given configuration for many models. Note, if you inherit from multiple BaseModel subclasses, like class MyModel(Model1, Model2), the non-default settings in the model_config attribute from the two models will be merged, and for any settings defined in both, those from Model2 will override those from Model1.

**Comment by KotlinIsland** [comment]
i find that very bizarre and unintuitive

**Comment by KotlinIsland** [comment]
@sydney-runkle was there any reasoning behind this pattern that you know of? what would it take to get it fixed?

**Comment by sydney-runkle** [comment]
@KotlinIsland,

I agree with you, this is a concerning pattern. Perhaps we could address in V3. I'll open and tag it as such.
